Source/1.3/Events/EventManager.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Empire_Rewritten.Controllers;
+using Empire_Rewritten.Settlements;
+using RimWorld;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+
+namespace Empire_Rewritten.Events
+{
+    public class EventManager : IExposable
+    {
+
+
+        private static Random rand = new Random();
+        private static List<EventDef> cachedEvents = new List<EventDef>();
+        private static int lastDay;
+        public EventManager()
+        {
+            UpdateController.CurrentWorldInstance.AddUpdateCall(DoRandomEventOnRandomEmpire,ShouldFireEvent);
+            cachedEvents = DefDatabase<EventDef>.AllDefsListForReading;
+        }
+
+        public static bool ShouldFireEvent()
+        {
+            int passedDays = GenDate.DaysPassed;
+
+            if (passedDays - lastDay > 15)
```

Comment:
The "arbitrary" 15 here should be linked somewhere with some constant that defines "turntime", otherwise if we ever change how long a "turn" is, this might desync and cause issues like ones we have in the current empire mod